lib/msf/core/post/hardware/automotive/uds.rb

```diff
@@ -441,6 +442,62 @@ def clear_dtcs(bus, srcId, dstId, opt={})
     client.automotive.send_isotp_and_wait_for_response(bus,srcId, dstId, [0x04], opt)
   end
 
+  ### Mode $07 ###
+
+  #
+  # Retrieves the Frozen Diagnostic Trouble Codes (DTCs)
+  #
+  # @param bus [String] unique CAN bus identifier
+  # @param srcId [Integer] Integer representation of the Sending CAN ID
+  # @param dstId [Integer] Integer representation of the receiving CAN ID
+  # @param opt [Hash] Additional options to be passed to automotive.send_isotp_and_wait_for_response
+  # @param opt [Hash] Additional options to be passed to automotive.send_isotp_and_wait_for_response
+  #
+  # @return [Array] Array of DTCs
+  def get_frozen_dtcs(bus, srcId, dstId, opt={})
+    dtcs = []
+    if not client.automotive
+      print_error("Not an automotive hwbridge session")
+      return {}
+    end
+    srcId = srcId.to_s(16)
+    dstId = dstId.to_s(16)
+    bus = client.automotive.active_bus if not bus
+    if not bus
+      print_line("No active bus, use 'connect' or specify bus via the options")
+      return {}
+    end
+    data = client.automotive.send_isotp_and_wait_for_response(bus,srcId, dstId, [0x07], opt)
+    return [] if data == nil
+    if data.has_key? "error"
+      print_error("UDS ERR: #{data["error"]}")
```

Comment:
What do you think about using a single quote inside the string interpolation here?

`print_error("UDS ERR: #{data['error']}")`